leaf-ui/rappid-extensions/ConfigInspector.js

```diff
@@ -0,0 +1,308 @@
+     */
+    updateHighlight : function(){
+        this.render();
+    }
+});
+
+/**
+ * View for dropdown result menu under configurations
+ * 
+ * {ResultCollection} collection
+ */
+var ResultsDropdown = Backbone.View.extend({
+    collection: ResultCollection,
+
+    /** Pass in a reference to parent configuration on intialization */
+    initialize: function(options){
+        this.config = options.config;
+    },
+
+    template: [
+    '<div class="dropdown-container">',
+    '</div>'].join(''),
+    
+    /**
+     * Resets listeners and resets template
+     * Then adds all results associated with the configuration
+     */
+    render: function() {
+        this.stopListening();
+        this.listenTo(this.collection, 'add', this.loadResult, this);
+        this.$el.html(_.template(this.template)());
+        this.collection.forEach(result => {this.loadResult(result)});
+        return this;
+    },
+
+    /** Adds new ResultView to dropdown container */
+    loadResult : function(result) {
+        var view = new ResultView({model: result, config: this.config});
+        this.$('.dropdown-container').append(view.render().el);
+    },
+})
+
+/**
+ * View for each configuration in the analysis configuration sidebar
+ * Also containers ResultsDropdown as a inner view
+ * 
+ * {ConfigModel} model
+ */
+var Config = Backbone.View.extend({
+    model: ConfigModel,
+
+    /** Create and render dropdown inner view, set listeners */
+    initialize: function(){
+        this.innerView = new ResultsDropdown({collection:this.model.get("results"), config: this.model});
+        this.innerView.render();
+        this.model.on('destroy', this.remove, this);
+        this.model.on('change:selected', this.updateHighlight, this);
+        this.model.on('change:name', this.renderName, this);
+    },
+
+    template: ['<script type="text/template" id="item-template">',
+    '<div class="analysis-configuration" id="<%= name %>">',
+        '<button class="config-elements" <% if (selected) { %> style="background-color:#A9A9A9;" <%} %> >',
+        '<%= name %> </button>',
+        '<input class="config-input" value="<%- name %>" style="display:none"></input>',
+        '<div id="config-buttons" style="position:absolute; display:inline-block">',
+        '<button class="deleteconfig-button">',
+            '<i id="garbage-icon" class="fa fa-trash-o" aria-hidden="true"></i>' +
+        '</button>',
+        '<button class="dropdown-button">',
+            '<i id="drop-icon" class="fa fa-caret-up fa-2x" style="cursor:pointer;"></i>',
+        '</button>',
+        '</div>',
+       '</div>',
+       '</script>'].join(''),
+
+    events: {
+        'click .config-elements': 'switchConfig',
+        'click .deleteconfig-button': 'removeConfig',
+        'click .dropdown-button' : 'toggleDropdown',
+        'dblclick .config-elements': 'rename',
+        'blur .config-input': 'setConfigName',
```

Comment:
Does this listener check if user is still typing?

Reply:
Are you referring to the .keyup event? It is triggered on any keyup event while the user is typing in the config-input, but the function itself is only used if the user presses the enter key to "escape" editing and save the new name.

Reply:
Should I add a more clarifying comment?

Reply:
I think it's fine!